### PR TITLE
ISPN-8326 Use regular methods to skip tests on some OSes

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-8326

Replaces https://github.com/infinispan/infinispan/pull/5452

### CODE REVIEW

File: commons-test/src/main/java/org/infinispan/commons/test/skip/SkipJunit.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Use as a {@code @Rule} or {@code @ClassRule} to skip all methods in a class on some OSes.
+ */
+public class SkipJunit implements TestRule {
+   private OS[] oses;
```
Comment (anistor) [inline commons-test/src/main/java/org/infinispan/commons/test/skip/SkipJunit.java]: I'd make it final.

File: commons-test/src/main/java/org/infinispan/commons/test/skip/SkipUtil.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.infinispan.commons.test.skip;
+
+class SkipUtil {
+   static OS getOs() {
```
Comment (anistor) [inline commons-test/src/main/java/org/infinispan/commons/test/skip/SkipUtil.java]: We could remove this class and move the getOs method inside OS enum so ```SkipUtil.getOs()``` becomes ```OS.get()```.

File: commons-test/src/main/java/org/infinispan/commons/test/skip/SkipTestNG.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Allows to skip a test on certain Operation Systems.
+ */
+public class SkipTestNG {
+   /**
+    * Use within a {@code @Test} method to skip that method on some OSes.
+    * Use in a {@code @BeforeClass} method to skip all methods in a class on some OSes.
+    */
+   public static void skipOnOS(OS... oses) {
+      OS currentOs = SkipUtil.getOs();
+      if (Arrays.asList(oses).indexOf(currentOs) >= 0) {
```
Comment (anistor) [inline commons-test/src/main/java/org/infinispan/commons/test/skip/SkipTestNG.java]: indexof -> contains

**Comment by danberindei** [comment]
Done @anistor

**Comment by anistor** [comment]
Integrated. Thanks @danberindei !

**Pull request merged**